src/makeSchema.ts

```diff
@@ -64,13 +65,61 @@ generateSchema.withArtifacts = async (
   schemaTypes: string
   tsTypes: string
 }> => {
-  const { schema, missingTypes, finalConfig } = makeSchemaInternal(config)
+  const { schema, missingTypes, finalConfig, hasSDLDirectives } = makeSchemaInternal(config)
   const typegenConfig = resolveTypegenConfig(finalConfig)
   const { schemaTypes, tsTypes } = await new TypegenMetadata(typegenConfig).generateArtifactContents(
     schema,
-    typeFilePath
+    typeFilePath,
+    hasSDLDirectives
   )
   assertNoMissingTypes(schema, missingTypes)
   runAbstractTypeRuntimeChecks(schema, finalConfig.features)
   return { schema, schemaTypes, tsTypes }
 }
+
+/** Builds the schema, we may return more than just the schema from this one day. */
+export function makeSchemaInternal(config: SchemaConfig) {
+  const builder = new SchemaBuilder(config)
+  builder.addTypes(config.types)
+  const {
+    finalConfig,
+    typeMap,
+    missingTypes,
+    schemaExtension,
+    onAfterBuildFns,
+    customDirectives,
+    schemaDirectives,
+    hasSDLDirectives,
+  } = builder.getFinalTypeMap()
+  const { Query, Mutation, Subscription } = typeMap
+
+  /* istanbul ignore next */
+  if (!isObjectType(Query)) {
+    throw new Error(`Expected Query to be a objectType, saw ${Query.constructor.name}`)
+  }
+  /* istanbul ignore next */
+  if (Mutation && !isObjectType(Mutation)) {
+    throw new Error(`Expected Mutation to be a objectType, saw ${Mutation.constructor.name}`)
+  }
+  /* istanbul ignore next */
+  if (Subscription && !isObjectType(Subscription)) {
+    throw new Error(`Expected Subscription to be a objectType, saw ${Subscription.constructor.name}`)
+  }
+
+  const schema = new GraphQLSchema({
+    query: Query,
+    mutation: Mutation,
+    subscription: Subscription,
+    types: objValues(typeMap),
+    extensions: {
+      ...config.extensions,
+      nexus: schemaExtension,
+    },
+    directives: [...specifiedDirectives, ...Object.values(customDirectives)],
+    ...schemaDirectives,
```

Comment:
I have a TS error in editor for that line `Property 'operationTypes' is missing in type 'ScalarTypeDefinitionNode' but required in type 'SchemaDefinitionNode'`

Reply:
I feel this is misleading for future changes, this returns an AST node. If some would need to override another property in the `astNode` they might override the `astNode` set by `schemaDirectives`.

Reply:
Yeah agreed, meant to come back to this / out this as a TODO